integration/tests/access/cohort3/execution_data_pruning_test.go

```diff
@@ -0,0 +1,292 @@
+	anEds := s.nodeExecutionDataStore(accessNode)
+
+	// setup storage objects needed to get the execution data id
+	anDB, err := accessNode.DB()
+	require.NoError(s.T(), err, "could not open db")
+
+	anHeaders := badger.NewHeaders(metrics, anDB)
+	anResults := badger.NewExecutionResults(metrics, anDB)
+
+	// start an execution data service using the Observer Node's execution data db
+
+	onEds := s.nodeExecutionDataStore(observerNode)
+	// setup storage objects needed to get the execution data id
+	onDB, err := observerNode.DB()
+	require.NoError(s.T(), err, "could not open db")
+
+	onResults := badger.NewExecutionResults(metrics, onDB)
+
+	s.checkResults(anHeaders, anResults, onResults, anEds, onEds)
+}
+
+// waitUntilExecutionDataForBlockIndexed waits until the execution data for the specified block height is indexed.
+// It subscribes to events from the start height and waits until the execution data for the specified block height is indexed.
+func (s *ExecutionDataPruningSuite) waitUntilExecutionDataForBlockIndexed(waitingBlockHeight uint64) {
+	observerNode := s.net.ContainerByName(s.observerNodeName)
+
+	grpcClient, err := s.getGRPCClient(observerNode.Addr(testnet.GRPCPort))
+	s.Require().NoError(err)
+
+	// creating execution data api client
+	client, err := getClient(fmt.Sprintf("localhost:%s", observerNode.Port(testnet.ExecutionStatePort)))
+	s.Require().NoError(err)
+
+	// pause until the observer node start indexing blocks,
+	// getting events from 2-nd block to make sure that 1-st block already indexed, and we can start subscribing
+	s.Require().Eventually(func() bool {
+		_, err := grpcClient.GetEventsForHeightRange(s.ctx, &accessproto.GetEventsForHeightRangeRequest{
+			Type:                 sdk.EventAccountCreated,
+			StartHeight:          2,
+			EndHeight:            2,
+			EventEncodingVersion: entities.EventEncodingVersion_CCF_V0,
+		})
+
+		return err == nil
+	}, 2*time.Minute, 10*time.Second)
+
+	// subscribe on events till waitingBlockHeight to make sure that execution data for block indexed till waitingBlockHeight and pruner
+	// pruned execution data at least once
+	// SubscribeEventsFromStartHeight used as subscription here because we need to make sure that execution data are already indexed
+	stream, err := client.SubscribeEventsFromStartHeight(s.ctx, &executiondata.SubscribeEventsFromStartHeightRequest{
+		EventEncodingVersion: entities.EventEncodingVersion_CCF_V0,
+		Filter:               &executiondata.EventFilter{},
+		HeartbeatInterval:    1,
+		StartBlockHeight:     0,
+	})
+	s.Require().NoError(err)
+	eventsChan, errChan, err := SubscribeHandler(s.ctx, stream.Recv, eventsResponseHandler)
+	s.Require().NoError(err)
+
+	duration := 3 * time.Minute
+	timer := time.NewTimer(duration)
+	defer timer.Stop()
+
+	for {
+		select {
+		case err := <-errChan:
+			s.Require().NoErrorf(err, "unexpected %s error", s.observerNodeName)
+		case event := <-eventsChan:
+			if event.Height > waitingBlockHeight {
```

Comment:
`waitUntilExecutionDataForBlockIndexed` seems to imply it will return when it reaches the height. Does it need to return after?
```suggestion
			if event.Height >= waitingBlockHeight {
```

Reply:
[2ec6bb6](https://github.com/onflow/flow-go/pull/6217/commits/2ec6bb67785a9dfcaab4dc1843b0ac3e4728f576)